### ISSUE TITLE
Inconsistent outputs due to timezone

### ISSUE THREAD

**Issue by benjaminRomano** [bazelbuild/rules_kotlin#301]
**Background**
I'm seeing cache misses due to the kotlin jars have different file timestamps. It appears, the kotlin rules depend on the [system default timezone](https://github.com/bazelbuild/rules_kotlin/blob/a4e902d33504b9a3fe2c46901c67973a74b50f40/src/main/kotlin/io/bazel/kotlin/builder/utils/jars/JarHelper.kt#L240) which may not be consistent across machines. 

**Suggested Fix**
It seems that setting the timezone to "UTC" should resolve this issue, but I still saw inconsistencies in timestamps between CI and my local machine. I haven't had a chance to debug further, but one thing I was doing differently was depending on kotlin rules using `git_repository` instead of building the release artifact.

**Workaround**
I'm able to workaround this issue by setting my machines timezone to match our CI builder's timezone UTC for the time being.

**Comment by davidsantiago** [comment]
It looks like there's an old PR (#167) that touches the same line you've identified, but it looks like they're solving the problem differently. Does this PR look like it would fix the issue for you?

**Comment by benjaminRomano** [comment]
Interesting. I found the original commit that introduced this change [here](https://github.com/bazelbuild/bazel/issues/4614#issuecomment-364704590). It seems including the timezone is intentional.

**Comment by davidsantiago** [comment]
Do I understand from your comment that you updated the code in rules_kotlin to specify UTC as the time zone and it still had inconsistencies? Or did you mean updating the timezone of your local machine?

**Comment by benjaminRomano** [comment]
Sorry for slow response. I updated the code to use a standard timezone, UTC; however, this was not the correct solution as it causes the timezone information to be encoded within the zip.

I tracked down the root cause when investigating a similar issue within Bazel's android rules. The problem is JDK 8 appears to have a bug when using unix epoch. It will encode the system's timezone into the zip archive making it non-reproducible. There's a good explanation of the issue within Gradle's archiving source code: https://github.com/gradle/gradle/blob/master/subprojects/core/src/main/java/org/gradle/api/internal/file/archive/ZipCopyAction.java#L42-L56
